src/test/java/de/metas/ui/web/notes/json/JSONNoteTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.ui.web.notes.json;
+
+import de.metas.util.JSONObjectMapper;
+import org.junit.jupiter.api.Test;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class JSONNoteTest
+{
+
+	@Test
+	void testSerialisationDeserialisation()
+	{
+		final JSONObjectMapper<JSONNote> jsonObjectMapper = JSONObjectMapper.forClass(JSONNote.class);
+
+		final JSONNote expectedNote = JSONNote.builder()
+				.createdBy("who created this?")
+				.created(ZonedDateTime.of(2020, 4, 22, 11, 11, 11, 0, ZoneId.of("UTC+3")))
+				.text("This is a test note.\nTra la la.")
+				.build();
+
+		final String json = jsonObjectMapper.writeValueAsString(expectedNote);
+		final JSONNote deserialisedRequest = jsonObjectMapper.readValue(json);
+		assertThat(deserialisedRequest).isEqualToIgnoringGivenFields(expectedNote);
```

Comment:
why not simply isEqualTo?

Reply:
I see that in the cases where i have `actual.a = "2"` (string) and `expected.a = 2L` (long), `isEqualToIgnoringGivenFields` will show the type of the field in the error message.

`isEqualTo` does not show type. it will only error out with a message like `expected 2 to equal 2`.